lib/ansible/utils/ssh_agent.py

```diff
@@ -0,0 +1,666 @@
+
+    @classmethod
+    def from_blob(cls, blob: memoryview | bytes) -> t.Self:
+        return cls(blob)
+
+
+class unicode_string(str, VariableSized):
+    def to_blob(self) -> bytes:
+        val = self.encode('utf-8')
+        if length := len(val):
+            return uint32(length).to_blob() + val
+        else:
+            return b""
+
+    @classmethod
+    def from_blob(cls, blob: memoryview | bytes) -> t.Self:
+        return cls(bytes(blob).decode('utf-8'))
+
+
+class KeyAlgo(str, VariableSized, enum.Enum):
+    RSA = "ssh-rsa"
+    DSA = "ssh-dss"
+    ECDSA256 = "ecdsa-sha2-nistp256"
+    SKECDSA256 = "sk-ecdsa-sha2-nistp256@openssh.com"
+    ECDSA384 = "ecdsa-sha2-nistp384"
+    ECDSA521 = "ecdsa-sha2-nistp521"
+    ED25519 = "ssh-ed25519"
+    SKED25519 = "sk-ssh-ed25519@openssh.com"
+    RSASHA256 = "rsa-sha2-256"
+    RSASHA512 = "rsa-sha2-512"
+
+    @property
+    def main_type(self):
+        match self:
+            case self.RSA:
+                return 'RSA'
+            case self.DSA:
+                return 'DSA'
+            case self.ECDSA256 | self.ECDSA384 | self.ECDSA521:
+                return 'ECDSA'
+            case self.ED25519:
+                return 'ED25519'
+            case _:
+                raise NotImplementedError(self.name)
+
+    def to_blob(self) -> bytes:
+        b_self = self.encode('utf-8')
+        return uint32(len(b_self)).to_blob() + b_self
+
+    @classmethod
+    def from_blob(cls, blob: memoryview | bytes) -> t.Self:
+        return cls(bytes(blob).decode('utf-8'))
+
+
+if HAS_CRYPTOGRAPHY:
+    _ECDSA_KEY_TYPE: dict[KeyAlgo, type[EllipticCurve]] = {
+        KeyAlgo.ECDSA256: SECP256R1,
+        KeyAlgo.ECDSA384: SECP384R1,
+        KeyAlgo.ECDSA521: SECP521R1,
+    }
+
+
+@dataclasses.dataclass
+class Msg:
+    def to_blob(self) -> bytes:
+        rv = bytearray()
+        for field in dataclasses.fields(self):
+            fv = getattr(self, field.name)
+            if isinstance(fv, SupportsToBlob):
+                rv.extend(fv.to_blob())
+            else:
+                raise NotImplementedError(field.type)
+        return rv
+
+    @classmethod
+    def from_blob(cls, blob: memoryview | bytes) -> t.Self:
+        args: list[t.Any] = []
+        for _field_name, field_type in t.get_type_hints(cls).items():
+            if isinstance(field_type, SupportsFromBlob):
+                fv, blob = field_type.consume_from_blob(blob)
+                args.append(fv)
+            else:
+                raise NotImplementedError(str(field_type))
+        return cls(*args)
+
+
+@dataclasses.dataclass(order=True, slots=True)
+class AgentLockMsg(Msg):
+    passphrase: binary_string
+
+
+@dataclasses.dataclass
+class PrivateKeyMsg(Msg):
+    @staticmethod
+    def from_private_key(private_key: CryptoPrivateKey):
+        match private_key:
+            case RSAPrivateKey():
+                pn: RSAPrivateNumbers = private_key.private_numbers()  # type: ignore[no-redef]
```

Comment:
Instead of type-ignoring, just use different variable names for things in the same scope.
```suggestion
                rsa_private_nums: RSAPrivateNumbers = private_key.private_numbers()
```
would make it so that MyPy wouldn't assume conflicting types. Also, the first occurrence probably doesn't need an ignore anyway.